### ISSUE TITLE
Hotloading: Async method marked as completed too early

### ISSUE THREAD

**Issue by DevulTj** [Facepunch/sbox-issues#7098]
### Describe the bug

I noticed it if you hotload sometimes (it's not 100%, but if you spam save on a file a few times you'll notice it on an empty project)

### To Reproduce

1. Open attached project
2. Open in play mode
3. Notice "Before Test" text in the console
4. If you hotload a few times, you will notice "After Test" will run before the 30 seconds timer is up. 

### Expected behavior

Should maintain state

### Media/Files

[async_hotloading_test.zip](https://github.com/user-attachments/files/17976391/async_hotloading_test.zip)

**Comment by DevulTj** [comment]
I'm also gonna note here that the same issue seems to happen when performing host migration - which makes sense. I've not looked too much into it though, I don't see how we'd make it work really.

**Comment by Metapyziks** [comment]
Thanks for the repro. Looks like Time.Now jumps occasionally, maybe Task.DelayInternal() continues in the wrong Time.Scope().

**Comment by Metapyziks** [comment]
Yeah we're pumping the main thread sync context continuation queue at a few points during hotload.

I think this is exposing a bigger problem, there will be continuations in there for both editor and game tasks, with different scene and / or time scopes. Either we should have separate SyncContexts for editor vs game, or capture things like the active scene when posting to a sync context. I'm going to mock up the latter because it should be a smaller change.
